### DISCUSSION TITLE
wire:navigate and browser's back button destroys flickity carousel

### DISCUSSION THREAD

**Discussion by patriktoth67** [livewire/livewire#6851] [7. Bugs (archived)] [answered]
### Livewire version

3.0.5

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

I'm using wire:navigate with full-page components, and on one page there is a flickity-carousel.

If I navigate away from it, then press the browsers back button, the carousel gets destroyed. 

Here is the code to reproduce:

web.php
```
<?php

use App\Livewire\Pages\{
    Test1,
    Test2
};

use Illuminate\Support\Facades\Route;

Route::get('/test1', Test1::class);
Route::get('/test2', Test2::class);
```

layouts/app.blade.php
```
<!DOCTYPE html>
<html lang="{{ str_replace('_', '-', app()->getLocale()) }}">

<head>
    <link rel="stylesheet" href="https://unpkg.com/flickity@2/dist/flickity.min.css">
    <script src="https://unpkg.com/flickity@2/dist/flickity.pkgd.min.js"></script>
    @livewireStyles
</head>

<body>
    <a href="/test1" wire:navigate>Test 1</a>
    <a href="/test2" wire:navigate>Test 2</a>
    {{ $slot }}
    @livewireScripts
</body>

</html>
```

Test1.php (Test2.php is the same, but renders the test2 blade file)
```
<?php

namespace App\Livewire\Pages;

use Livewire\Component;

class Test1 extends Component
{
    public function render()
    {
        return view('livewire.pages.test1')->layout('layouts.app'); // Renders livewire.pages.test2 for Test2.php
    }
}
```

livewire/pages/test1.blade.php
```
<div class="carousel">
    <p>1</p>
    <p>2</p>
    <p>3</p>
    <p>4</p>
    <p>5</p>
    <p>6</p>
    <p>7</p>
    <p>8</p>
</div>

<script>
    var flkty = new Flickity('.carousel');
</script>

```

livewire/pages/test2.blade.php
```
<div></div>
```

Opening the url to test1 (/test1) gives the following:
![image](https://github.com/livewire/livewire/assets/47533464/e8e190b2-54f4-498f-8e9b-e56d49f85ec5)

As you can see the carousel is initialized, all good.

After clicking the link to test2, then pressing the browsers back button, I get the following:
![image](https://github.com/livewire/livewire/assets/47533464/dd507fd7-cf4b-4e86-ad28-cc853a72eea2)

The carousel is no longer working, it has no items and looks bugged.

Removing wire:navigate from the links solves the issue.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by N1ebieski** [comment]
Probably related #5894

**Reply by patriktoth67** [reply, marked as answer]
Thank you, yes the solution is to add this:

```
document.addEventListener('livewire:navigating', () => {
        flkty.destroy();
    });
    ```
